di-ipv-cimit-stub/lambdas/stub-management/src/main/java/uk/gov/di/ipv/core/stubmanagement/service/impl/UserServiceImpl.java

```diff
@@ -35,40 +37,77 @@ public UserServiceImpl(ConfigService configService, CimitStubService cimitStubSe
     @Override
     public void addUserCis(String userId, List<UserCisRequest> userCisRequest) {
         List<CimitStubItem> cimitStubItems = cimitStubService.getCimitStubItems(userId);
-        if (!cimitStubItems.isEmpty()) {
-            throw new DataAlreadyExistException(
-                    "User already exists, instead try calling update api.");
-        }
         userCisRequest.forEach(
                 user -> {
-                    cimitStubService.persistCimitStub(
-                            userId,
-                            user.getCode(),
-                            getIssuanceDate(user.getIssuenceDate()),
-                            user.getMitigations());
+                    Optional<CimitStubItem> cimitStubItem =
+                            getUserIdAndCodeFromDatabase(cimitStubItems, user.getCode());
+                    if (cimitStubItem.isEmpty()) {
+                        cimitStubService.persistCimitStub(
+                                userId,
+                                user.getCode(),
+                                getIssuanceDate(user.getIssuanceDate()),
+                                user.getMitigations());
+                    } else {
+                        cimitStubItem
+                                .get()
+                                .setMitigations(
+                                        getUpdatedMitigationsList(
+                                                cimitStubItem.get().getMitigations(),
+                                                user.getMitigations()));
+                        cimitStubItem
+                                .get()
+                                .setIssuanceDate(getIssuanceDate(user.getIssuanceDate()));
+                        cimitStubService.updateCimitStub(cimitStubItem.get());
+                    }
                 });
         LOGGER.info("Inserted User CI data to the Cimit Stub DynamoDB Table.");
     }
 
     @Override
     public void updateUserCis(String userId, List<UserCisRequest> userCisRequest) {
+        if (userCisRequest.stream().allMatch(user -> StringUtils.isEmpty(user.getCode()))) {
```

Comment:
We'd be better erroring on *any* empty CI code in both the addUserCis and updateUserCis cases

Reply:
Done.